test/unit/Cardano/Wallet/PrimitiveSpec.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```

Comment:
Pretty-much yes. Ideally, we probably want something a bit more like Hedgehog were we define generators explicitly, and not via type-class. Because generators are kinda tailored to one or more tests. For instance, for those primitive, we do not care much about the shape of an address or a block header hash, that wouldn't be true of course if we were to test how addresses are serialized / unserialized!